### PR TITLE
feat: upgrade features project

### PR DESCRIPTION
**Pull request by petter-a**
# Upgrade Resources project
1. Change node version from 10 to 14
2. Upgrade packages in package.json
3. Add retention for API gateway logs for 2 weeks
4. Add AccessLogs for API gateway
5. Fixed legacy configuration for eslint

### CODE REVIEW

**Review by yo-l1982 — APPROVED** [review]
Works fine here.

File: services/gateway/root/serverless.yml [hunk 1]
```diff
@@ -12,11 +12,21 @@ provider:
...
+    metrics: true
 
   environment:
     stage: ${self:custom.stage}
     resourcesStage: ${self:custom.resourcesStage}
 
+  logs:
+    restApi:
+      fullExecutionData: false
+      executionLogging: true
```
Comment (karlknowit) [inline services/gateway/root/serverless.yml:26]: true is the default value for executionLogging

File: services/gateway/root/serverless.yml [hunk 2]
```diff
@@ -12,11 +12,21 @@ provider:
...
+      accessLogging: true
```
Comment (karlknowit) [inline services/gateway/root/serverless.yml:27]: true is the default value for accessLoggin

**Review by karlknowit — CHANGES_REQUESTED** [review]

File: services/gateway/root/serverless.yml (hunk 1 above)
Comment (yo-l1982) [reply services/gateway/root/serverless.yml:26]: It doesnt seem to be activated on the AWS dev environment and it should be if these are defaulted to true imo (as they were missing before).
Maybe we should keep them in this pr?

File: services/gateway/root/serverless.yml (hunk 2 above)
Comment (petter-a) [reply services/gateway/root/serverless.yml:27]: @karlknowit so they say.... but its not in practise

File: services/gateway/root/serverless.yml (hunk 1 above)
Comment (petter-a) [reply services/gateway/root/serverless.yml:26]: My experience with these settings are that they doesn't always "bite". It is a bit random. I agree that the documentation states that both execution logging and access logging are enabled by default but it did not appear before I actually enforced it. I will do another check of the Cloudformation template generated....

Comment (karlknowit) [reply services/gateway/root/serverless.yml:26]: We could keep them for clarification and to be sure that they values are actually set. I don't mind that

**Review by karlknowit — APPROVED** [review]

File: services/gateway/root/serverless.yml (hunk 2 above)
Comment (petter-a) [reply services/gateway/root/serverless.yml:27]: The logic in serverless seems to indicate that it indeed does enable access logging by default. However, it doesn't seem to work all the time. I will remove the override though...

const accessLogging = logs.accessLogging == null ? true : logs.accessLogging;

File: services/gateway/root/serverless.yml (hunk 1 above)
Comment (petter-a) [reply services/gateway/root/serverless.yml:26]: Decided to keep them for now. If nothing else, it doesn't harm

File: services/gateway/root/serverless.yml (hunk 2 above)
Comment (petter-a) [reply services/gateway/root/serverless.yml:27]: Decided to keep it for now as it doesn't cause any harm

**Pull request merged**